modules/accredited_representative_portal/app/controllers/accredited_representative_portal/v0/in_progress_forms_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+  module V0
+    class InProgressFormsController < ApplicationController
+      def update
+        form = InProgressForm.where(form_id:, user_uuid: @current_user.uuid).first_or_initialize
+        form.update!(form_data: params[:form_data] || params[:formData], metadata: params[:metadata])
+
+        render json: form, key_transform: :unaltered
+      end
+
+      def show
+        render json: in_progress_form&.data_and_metadata || {}
+      end
+
+      def destroy
+        raise Common::Exceptions::RecordNotFound, form_id if in_progress_form.blank?
+
+        in_progress_form.destroy
+        render json: in_progress_form, key_transform: :unaltered
+      end
+
+      private
+
+      def in_progress_form
+        @in_progress_form ||= form_for_user(form_id)
```

Comment:
`||=` won't memoize `nil` which will result here in repeated DB queries if someone thinks they are working with a memoized value returning function and are confident in invoking to it multiple times. It's risky to leave this as `||=`, so change it to something that can memoize `nil` by using `defined?`.

Reply:
Another option is to just use local variables rather than a memoized method. In my old age I might prefer the simplicity of that, but up to you.

Reply:
I think to memorize nil you'd just do:

```suggestion
        return @in_progress_form if defined?(@in_progress_form)
        
        @in_progress_form = form_for_user(form_id)
```